### PR TITLE
Cleanup and add a few new feature examples

### LINKED ISSUE #9: Remove use of AssertionHelper

### PR DESCRIPTION
**Pull request by rprouse**
I needed to demo some NUnit features and couldn't find ready examples, so I decided to update this repository. In the end, I added a few new features in a new project, added an overall solution so users can browse everything easily and updated the cake build. Need to see if all this still builds on CI.

I would like to start expanding out the examples in this project. We especially need coverage of the various attributes and options for data driven tests.

Fixes #9

### CODE REVIEW

**Comment by rprouse** [comment]
This also removes the obsolete `AssertionHelper` code. Sorry it is so huge, but it builds on AppVeyor and Travis, plus is mainly additive. It probably just needs a quick review.

File: DataDrivenTests/DataDrivenTestFixture.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        int _x;
+
+        public DataDrivenTestFixture(int x)
+        {
+            _x = x;
+        }
+
+        [Test]
+        public void TestArgurments()
```
Comment (jnm2) [inline DataDrivenTests/DataDrivenTestFixture.cs]: Typo `Argurments`

File: DataDrivenTests/DataDrivenTests.csproj [hunk 2]
```diff
@@ -0,0 +1,66 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\packages\NUnit3TestAdapter.3.9.0\build\net35\NUnit3TestAdapter.props" Condition="Exists('..\packages\NUnit3TestAdapter.3.9.0\build\net35\NUnit3TestAdapter.props')" />
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
```
Comment (jnm2) [inline DataDrivenTests/DataDrivenTests.csproj]: Sort feel like using the new csproj format is a better demo, but easy enough to do later.
I always use the .NET Standard class library template and swap out the TargetFramework value.

File: DataDrivenTests/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+//      Major Version
+//      Minor Version
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (jnm2) [inline DataDrivenTests/Properties/AssemblyInfo.cs]: I usually nuke AssemblyInfo files but it isn't in the way.

File: syntax/AssertSyntaxTests.cs [hunk 4]
```diff
@@ -821,6 +609,48 @@ public void ComplexTests()
         //    Assert.That(c, Is.All.Not);
         //}
         #endregion
+
+        #region Assumptions
+
+        [Test]
+        public void PostiveAssumption()
```
Comment (jnm2) [inline syntax/AssertSyntaxTests.cs]: Typo `Postive`

File: syntax/AssertSyntaxTests.cs [hunk 5]
```diff
@@ -821,6 +609,48 @@ public void ComplexTests()
...
+        {
+            Assume.That(true);
+
+            Assert.Pass("This will be hit because of the assumption.");
```
Comment (jnm2) [inline syntax/AssertSyntaxTests.cs]: 💭 `will be hit` could be `will execute` here and other places—more familiar term for me but YMMV.

File: syntax/AssertSyntaxTests.cs [hunk 6]
```diff
@@ -821,6 +609,48 @@ public void ComplexTests()
...
+
+            Assert.Fail("This will not be hit because of the assumption.");
+        }
+
+        #endregion
+
+        #region Warnings
+
+        [Test]
+        public void PostiveWarning()
```
Comment (jnm2) [inline syntax/AssertSyntaxTests.cs]: ⚠ Typo `Postive`

File: syntax/AssertSyntaxTests.cs [hunk 7]
```diff
@@ -821,6 +609,48 @@ public void ComplexTests()
...
+        {
+            Warn.If(true, "This will emit a warning");
+            Warn.Unless(false, "This will emit a warning");
```
Comment (jnm2) [inline syntax/AssertSyntaxTests.cs]: ℹ Most assertion messages nearby have end marks.

File: DataDrivenTests/DataDrivenTests.csproj (hunk 2 above)
Comment (rprouse) [reply DataDrivenTests/DataDrivenTests.csproj]: I want this to be buildable on as many versions of Visual Studio as possible. I am thinking of adding a .NET Core/Standard solution as an example of that.

**Comment by rprouse** [comment]
Thanks for the review @jnm2. I've updated everything based on your feedback except the project format and AssemblyInfo. Waiting on CI.

File: DataDrivenTests/DataDrivenTests.csproj (hunk 2 above)
Comment (jnm2) [reply DataDrivenTests/DataDrivenTests.csproj]: Good thinking.

**Comment by jnm2** [comment]
Sorry for the wait; was driving.

**Pull request merged**